src/protocol/prf_sharding/mod.rs

```diff
@@ -677,17 +692,20 @@ where
     F: PrimeField + ExtendableField,
 {
     let mut step: usize = 1 << BK::BITS;
-    let mut row_contribution = vec![value; 1 << BK::BITS];
+
+    assert!(breakdown_count <= 1 << BK::BITS);
+    let mut row_contribution = vec![value; breakdown_count];
```

Comment:
It *might* be more efficient to initialize the vector with zero values instead, but that means putting "value" at index 0 as an extra step.